### PR TITLE
Feature osw changes

### PR DESCRIPTION
**Pull request by MashB**
Readme updates

### CODE REVIEW

**Review by susrisha — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,20 +1,19 @@
 # Introduction 
-Data service micro-service helps TDEI system to query information specific to the OSW.
+Data service micro-service helps TDEI system to query & persist the information specific to the OSW.
```
Comment (rdevalap) [inline README.md:2]: This service helps in persisting & qyerying information related to OSW data.
Cannot query before persisting. That's the idea

File: README.md [hunk 2]
```diff
@@ -30,9 +29,18 @@ Step 1:
...
-An example of this is environment file is available [here](./env.example)
+## Local Postgresql database setup
+
+Step 1: Ensure all the environment variables are setup.
+
+Step 2: Ensure docker is installed on local system. 
+
+Step 3: Run below command which will setup Postgresql database and PgAdmin client console for postgresql database.
+
+```docker compose up from root directory```
```
Comment (rdevalap) [inline README.md:40]: ```docker compose up``` from root directory

File: README.md [hunk 3]
```diff
@@ -30,9 +29,18 @@ Step 1:
...
+
+## Build and Test
 
-#### Build and Test
 Follow the steps to install the node packages required for both building and running the application
 
 1. Install the dependencies. Run the following command in terminal on the same directory as `package.json`
```
Comment (rdevalap) [inline README.md:46]: same directory level as `package.json`

File: README.md [hunk 4]
```diff
@@ -41,8 +49,47 @@ Follow the steps to install the node packages required for both building and run
     ```
 2. To start the server, use the command `npm run start`
 3. The http server by default starts with 3000 port or whatever is declared in `process.env.PORT` (look at `index.ts` for more details)
-4. By default `get` call on `localhost:3000` gives a sample response
-5. Other routes include a `ping` with get and post. Make `get` or `post` request to `http://localhost:3000/ping`
+4. Health check available at path `health/ping` with get and post. Make `get` or `post` request to `http://localhost:3000/health/ping`
```
Comment (rdevalap) [inline README.md:52]: What to do with health check? What to expect after ping? What needs to be done if it didn't give the expected response?

File: README.md [hunk 5]
```diff
@@ -41,8 +49,47 @@ Follow the steps to install the node packages required for both building and run
...
+    B -->|Save| D(OSW Database)
+    B -->|Auth| E(Auth Service)
+    G(Gateway) -->|GET| B(Data Service)
+```
+
+- `Data service`, subscribes to `osw-validation` topic to listen to validation results of the osw file upload request.
+
+- `Data service`, authorizes the request via `Auth Service` 
+
+- If validation result is failed , Data service publishes the information to `osw-data` topic to update request status complete without persisting the information.
```
Comment (rdevalap) [inline README.md:72]: What "validation" are we talking about here?
We went directly from Authorization to "Validation result". Is it the auth validation? or what?

**Review by rdevalap — COMMENTED** [review]
@MashB @susrisha Some comments made. How can we incorporate them?

**Comment by MashB** [comment]
@rdevalap , I have taken care of feedback in latest PR https://github.com/TaskarCenterAtUW/TDEI-osw-datasvc-ts/pull/21 
Please ignore other changes, PR got merged with old open PR

**Pull request merged**